File: solutions/_448_find_all_numbers_disappeared_in_an_array/solution.java
package _448_find_all_numbers_disappeared_in_an_array;

import java.util.*;

class Solution {

  public List<Integer> findDisappearedNumbers(int[] nums) {
    List<Integer> list = new ArrayList<>();
    for (int i = 0; i < nums.length; i++) {
      int correctIdx = nums[i] - 1;
      if (nums[i] != nums[correctIdx]) {
        int temp = nums[i];
        nums[i] = nums[correctIdx];
        nums[correctIdx] = temp;
        i--;
      }
    }
    for (int i = 0; i < nums.length; i++) {
      if (nums[i] - 1 != i) list.add(i + 1);
    }
    return list;
  }
}
/*
https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/
*/

File: solutions/_448_find_all_numbers_disappeared_in_an_array/solution.py
from typing import List


class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        i = 0
        while i < len(nums):
            corrext_idx = nums[i] - 1
            if nums[i] != nums[corrext_idx]:
                nums[corrext_idx], nums[i] = nums[i], nums[corrext_idx]
            else:
                i += 1

        return [i + 1 for i, n in enumerate(nums) if n != i + 1]


# https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/
